src/views/TechnicalAssistance/RequestForm/index.tsx

```diff
@@ -0,0 +1,188 @@
+  const requestType = location.state?.requestType as TRBRequestType;
+
+  const { id, step, view } = useParams<{
+    id: string;
+    step?: string;
+    view?: string;
+  }>();
+
+  const [create, createResult] = useMutation<
+    CreateTrbRequest,
+    CreateTrbRequestVariables
+  >(CreateTrbRequestQuery);
+
+  const [get, getResult] = useLazyQuery<GetTrbRequest, GetTrbRequestVariables>(
+    GetTrbRequestQuery
+  );
+
+  // const [update, updateResult] = useMutation<
+  //   UpdateTrbRequest,
+  //   UpdateTrbRequestVariables
+  // >(UpdateTrbRequestQuery);
+  // todo update, updateResult
+
+  // Assign request data from the first available query response
+  // Prioritize by latest type: update, get, create
+  // eslint-disable-next-line camelcase
+  const request: CreateTrbRequest_createTRBRequest | undefined =
+    getResult.data?.trbRequest || createResult.data?.createTRBRequest;
+  // const data = useMemo(() => {
+  //   if (getResult.data) {
+  //     return getResult.data.trbRequest;
+  //   }
+  //   if (createResult.data) {
+  //     return createResult.data.createTRBRequest;
+  //   }
+  //   return undefined;
+  // }, [getResult, createResult]);
+  // console.log('data', request);
+
+  useEffect(() => {
+    // Create a new request if `id` is new and go to it
+    if (id === 'new') {
+      // Request type must be defined for new requests
+      if (requestType) {
+        if (!createResult.called) {
+          // Create the new request
+          create({ variables: { requestType } }).then(res => {
+            // Update the url with the request id
+            if (res.data) {
+              history.push(`/trb/requests/${res.data.createTRBRequest.id}`);
+            }
+          });
+        }
+      }
+      // Redirect to the start if there's no request type
+      else history.push('/trb/start');
+    }
+    // Fetch request data if not new
+    else if (!request && !createResult.called && !getResult.called) {
+      get({ variables: { id } });
+    }
+    // Get or create request was successful
+    // Continue any other effects with request data
+    else if (request) {
+      // Check step param, redirect to step 1 if invalid
+      if (!step?.match(RegExp(`[1-${formStepComponents.length}]|(done)`))) {
+        history.push(`/trb/requests/${id}/1`);
+      }
+    }
+  }, [
+    create,
+    createResult,
+    get,
+    getResult,
+    history,
+    id,
+    request,
+    requestType,
+    step
+  ]);
+
+  const defaultBreadcrumbs = useMemo(
+    () => (
+      <Breadcrumbs
+        items={[
+          { text: t('heading'), url: '/trb' },
+          { text: t('breadcrumbs.taskList'), url: '/trb/task-list' },
+          { text: t('requestForm.heading') }
+        ]}
+      />
+    ),
+    [t]
+  );
+
+  if (getResult.loading || createResult.loading) {
+    return <PageLoading />;
+  }
+
+  if (getResult.error || createResult.error) {
+    return <NotFoundPartial />;
+  }
```

Comment:
I don't believe these renderings should belong here.  I could be wrong, but I don't think we are rendering the request name in the summary box, nor any data really.  Might not even need a loading/error render.  I think the loading render should appear in child form components, and the summary box should always be rendered as it is static text

Reply:
@patrickseguraoddball I think these checks in this base view is ok. Basically the base component view renders on a valid existing request (id param) or one created successfully. If any of those error, is it actually necessary to still render the "summary box"/ form header? Users would still have to back out anyway.

Form step components and other subcomponents with any of their own queries are still able to handle further errors/loading inwards.

Reply:
I think it's a little unpleasant user experience cause it causes a screen flicker.  It's pretty noticeable.  I think it would be better to render the static text, and on the rare occasion an error occurs, it would render an error page instead.